File: 백준/Silver/2606. 바이러스/바이러스.py
import sys

N = int(sys.stdin.readline().rstrip())
M = int(sys.stdin.readline().rstrip())

graph = [[] for _ in range(N+1)]

for _ in range(M):
    a, b = map(int,sys.stdin.readline().split())
    graph[a].append(b)
    graph[b].append(a)


visitied = [0] * (N+1)

def dfs(n):
    visitied[n] = 1

    for nxt in graph[n]:
        if not visitied[nxt]:
            dfs(nxt)

dfs(1)
print(sum(visitied)-1)

File: 백준/Silver/2606. 바이러스/바이러스.java
import java.io.*;
import java.util.*;

class Main {
  static List<List<Integer>> graph;
    static boolean[] visited;
    static int n, m;
    static int count = 0;


    public static void main (String[] args) throws IOException{
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

    n = Integer.parseInt(br.readLine());

    m = Integer.parseInt(br.readLine());

    graph = new ArrayList<>();
    for (int i = 0; i <= n; i++) {
      graph.add(new ArrayList<>());
    }

    for (int i = 0; i < m; i++) {
      String[] st = br.readLine().split(" ");
      
      int a = Integer.parseInt(st[0]);
      int b = Integer.parseInt(st[1]);
      graph.get(a).add(b);
      graph.get(b).add(a);
  
    }
    visited = new boolean[n+1];

    dfs(1);

    System.out.println(count);
  }static void dfs(int node) {
    visited[node] = true;

    for (int next : graph.get(node)) {
      if (!visited[next]) {
        count++;
        dfs(next);
      }
    }
  }
}
